javascript/javascript1/week3/CactusIO-interactive.js

```diff
@@ -0,0 +1,75 @@
+// -------------------------------------------------------------------CactusIO-interactive (Smart phone usage app)
+// ----------------------------------------------------------------------------Adding an activity
+const activities = [];
+
+function addActivity(date=new Date().toDateString(), activity=null, duration=null) {
+    let obj = {
+    'date' : date,// //date : date=new Date().toLocaleString()
+    'activity' : activity,
+    'duration' : duration
+    };
+    if (date != undefined && activity != null && duration != null) {
+        activities.push(obj);
+        console.log(activities);
+    }
+}
+
+addActivity();
+addActivity('Thu Dec 07 2021', "Youtube");
+addActivity('Thu Dec 07 2021', "Facebook", 45);
+addActivity(undefined, "Zoom", 20);
+addActivity(undefined, "Youtube", 30);
+addActivity(undefined, "Facebook", 45);
+addActivity(undefined, "Zoom", 20);
+
+// activities should now look like this
+// [{
+//     date: '23/7-18',
+//     activity: 'Youtube',
+//     duration: 30,
+// }]
+
+// ------------------------------------------------------------------------------------Show my status
+// ---------------------------------------------------------------------------------------------Usage limit
+
+function showStatus(activities){
+  let sumMin = 0;
+  let sumActivities = 0;
+  if (activities.length === 0) {
+    console.log(`Add some activities before calling showStatus`);
```

Comment:
nit:
you can do following trick
```javascript
if (activities.length === 0) {
    console.log(`Add some activities before calling showStatus`);
    return
}
activities.forEach((item, i) => {
// and so on

```
It's pretty popular trick to make code bit simplified.
If `return` statement is present in that place `else` is not required